tests/components/nws/test_weather.py

```diff
@@ -152,46 +115,68 @@
     assert forecast is None
 
 
-async def test_error_station(hass):
+async def test_error_station(hass, mock_simple_nws):
     """Test error in setting station."""
-    with patch("homeassistant.components.nws.weather.SimpleNWS") as mock_nws:
-        instance = mock_nws.return_value
-        instance.station = "ABC"
-        instance.set_station.side_effect = aiohttp.ClientError
-        instance.update_observation.return_value = mock_coro()
-        instance.update_forecast.return_value = mock_coro()
-        instance.observation = None
-        instance.forecast = None
-        await async_setup_component(hass, "weather", MINIMAL_CONFIG)
 
-        state = hass.states.get("weather.abc")
-        assert state is None
+    instance = mock_simple_nws.return_value
+    instance.set_station.side_effect = aiohttp.ClientError
+
+    assert await async_setup_component(hass, nws.DOMAIN, MINIMAL_CONFIG) is True
+    await hass.async_block_till_done()
+
+    assert hass.states.get("weather.abc_hourly") is None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5329046)